use_cases/eluc/prescriptors/prescriptor.py

```diff
@@ -0,0 +1,281 @@
+from typing import Any
+from typing import Dict
+from typing import List
+
+import pandas as pd
+import numpy as np
+
+from esp_sdk.esp_evaluator import EspEvaluator
+
+from data import constants
+from data.eluc_data import ELUCEncoder
+
+def reco_to_context_actions(reco_df: pd.DataFrame, 
+                            encoded_context_df: pd.DataFrame, 
+                            data_encoder: pd.DataFrame) -> pd.DataFrame:
+    """
+    Converts a dataframe containing recommended land use proportions to a dataframe containing
+    the context and prescribed actions.
+    """
+    # This is gacky but has to happen sooner or later
+    reco_df = reco_df.reset_index(drop=True)
+    encoded_context_df = encoded_context_df.reset_index(drop=True)
+
+    context_df = data_encoder.decode_as_df(encoded_context_df)
+
+    # Linear scaling is implemented here:
+    # Do ReLU here since we no longer softmax
+    reco_df = reco_df.clip(0, None)
+    # If all outputs 0, set to be uniform
+    reco_df[reco_df.sum(axis=1) == 0] = 1 # Could be any positive constant, 1 for simplicity
+    prescribed_total_df = reco_df.sum(axis=1)
+    prescribed_total_df = prescribed_total_df.replace(0, 1)
+    # Since we are no longer using softmax, do a linear scaling
+    reco_df = reco_df.div(prescribed_total_df, axis=0)
+
+    # Scale encoded_reco_df to context_df minus primf/primn
+    # Multiply proportions by sum of non primn/primf cols
+    reco_df = reco_df.mul(context_df[constants.RECO_COLS].sum(axis=1), axis=0)
+
+    # Compute the diff
+    # Note: the index need to match in order to subtract. Otherwise we get NaN
+    prescribed_actions_df = reco_df[constants.RECO_COLS].reset_index(drop=True) - context_df[constants.RECO_COLS].reset_index(drop=True)
+
+    # Rename the columns to match what the predictor expects
+    prescribed_actions_df = prescribed_actions_df.rename(constants.RECO_MAP, axis=1)
+    prescribed_actions_df[constants.NO_CHANGE_COLS] = 0
+    
+    # Aggregate the context and actions dataframes.
+    context_actions_df = pd.concat([context_df,
+                                            prescribed_actions_df[constants.DIFF_LAND_USE_COLS]],
+                                            axis=1)
+    
+    return context_actions_df
+
+class UnileafPrescriptor(EspEvaluator):
```

Comment:
We should rename this class too in next PR. Overall, remove references to ESP and Unileaf